pyteal/ast/router.py

```diff
@@ -508,19 +496,28 @@ def __init__(
         name: str,
         bare_calls: BareCallActions | None = None,
         descr: str | None = None,
+        *,
+        clear_state: Optional[ActionType] = None,
     ) -> None:
         """
         Args:
             name: the name of the smart contract, used in the JSON object.
             bare_calls: the bare app call registered for each on_completion.
             descr: a description of the smart contract, used in the JSON object.
+            clear_state: an expression describing the behavior of clear state program. This
+                expression will be the entirety of the clear state program; no additional code is
+                inserted by the Router. If not provided, the clear state program will always reject.
         """
 
         self.name: str = name
         self.descr = descr
 
         self.approval_ast = ASTBuilder()
-        self.clear_state_ast = ASTBuilder()
+        self.clear_state: Expr = (
```

Comment:
I believe this is backwards compatible in the sense that if we have a program written today with default behavior, it will generate an `int 0` for its clear. However, it is arguable that even in _this_ default case, we still want a _NEW_ behavior and produce `int 1` instead for the clear program.

And assuming we agree with this point, then unfortunately, the parameter will need to become mandatory because of the lack of backwards compatibility. So keeping it as kwarg-only param, we would need to have something like:

```python
assert clear_state, "... something similar to the other error messages above ..."
```

In either case, I encourage the creation of a tiny unit test (assuming it doesn't already exist) that will assert that the simplest compiling PyTeal program generates the approval and clear programs that we would like to generate.

Reply:
Upon further reflection, I changed my mind.... `int 0` for the simplest possible clear program still makes sense because there is no state that is changing inside of the program and that needs to be preserved.

Reply:
Thought for a bit, we still keep the backwards compatible way, right?

One question: even one call a CSP with `int 0`, then it failed, it still get its state cleared, right?